### DISCUSSION TITLE
Lesson 6 cast call command giving 0x as answer

### DISCUSSION THREAD

**Discussion by msyuvaraj** [Cyfrin/foundry-full-course-cu#956] [General]
Hi , Im  having this  issue getting 0x while cast call retrive( )   in lesson 6.
i did append the `--broadcast` commad while deploying and it was succefully deployed

this is the command i used for deploying
`$ forge script script/DeploySimpleStorage.s.sol --rpc-url $RPC_URL --broadcast --private-key $PRIVATE_KEY`

this is the command i used for interacting i.e to store the number
`$ cast send 0x2279B7A0a67DB372996a5FaB50D91eAA73d2eBe6  "store(uint256)" 145 --rpc-url $RPC_URL --private-key $PRIVATE_KEY`

and this is what i got when cast called

`$ cast call 0x2279B7A0a67DB372996a5FaB50D91eAA73d2eBe6 "retrieve()"`

`0x`

could someone please help me?

**Comment by 0xusmanf** [comment]
You also need to specify the return type along with `--rpc-url` like in the below example.
```shell
 cast call 0x2279B7A0a67DB372996a5FaB50D91eAA73d2eBe6 "retrieve()(uint256)" --rpc-url $RPC_URL
```

**Reply by msyuvaraj** [reply]
Tqsm @usmanfarooq91 its working.
And if you dont mind can you also please help me with another issue.

while cast send for the function addPerson( ) im getting the following error.I have attached the command below.
is there anything im missing here..? 

`$ cast send 0x2279B7A0a67DB372996a5FaB50D91eAA73d2eBe6  "addPerson(string,uint256)" '["yuvaraj",123]' --rpc-url $RPC_URL --private-key $PRIVATE_KEY 
Error: 
Invalid data`

**Reply by 0xusmanf** [reply]
You don't have to enclose input data into anything, just do as follows.

```shell
cast send 0x2279B7A0a67DB372996a5FaB50D91eAA73d2eBe6  "addPerson(string,uint256)" "yuvaraj",123 --rpc-url $RPC_URL --private-key $PRIVATE_KEY
```

**Reply by msyuvaraj** [reply]
Thnx for your help @usmanfarooq91 .
`cast send 0x2279B7A0a67DB372996a5FaB50D91eAA73d2eBe6  "addPerson(string,uint256)" {"yuvaraj",123} --rpc-url $RPC_URL --private-key $PRIVATE_KEY`

this command worked out for me.
i just placed the input in a flower braces just like an object.
